File: files/300-399/345/CH-345.py
import pandas as pd
import numpy as np

path = "300-399/345/CH-345 Pattern Recongnition.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=2, nrows=6)
test = pd.read_excel(path, usecols="C", skiprows=2, nrows=6)

input["Count"] = input["Pattern"].apply(
    lambda x: sum(a != b for a, b in zip(x, x[1:]))
)

File: files/300-399/345/CH-345.R
library(tidyverse)
library(readxl)
library(charcuterie)

path <- "300-399/345/CH-345 Pattern Recongnition.xlsx"
input <- read_excel(path, range = "B3:B8")
test <- read_excel(path, range = "C3:C8")

result = input %>%
  mutate(Pattern = map(Pattern, chars)) %>%
  mutate(Count = map_int(Pattern, ~ sum(.x[-length(.x)] != .x[-1])))

# difference in 3 example.
